### PR TITLE
Translation of optuna.trial

### PR DESCRIPTION
**Pull request by chenghuzi**

### CODE REVIEW

File: source/locale/zh_CN/LC_MESSAGES/reference/generated/optuna.trial.FixedTrial.po [hunk 1]
```diff
@@ -23,7 +23,7 @@ msgstr ""
 
 #: of optuna.trial.FixedTrial:1
 msgid "A trial class which suggests a fixed value for each parameter."
-msgstr ""
+msgstr "为每一个参数提供固定值 suggestion 的 trial 类."
```
Comment (belldandyxtq) [inline source/locale/zh_CN/LC_MESSAGES/reference/generated/optuna.trial.FixedTrial.po]: 为每一个参数提供唯一固定值的 trial 类.

File: source/locale/zh_CN/LC_MESSAGES/reference/generated/optuna.trial.FixedTrial.po [hunk 2]
```diff
@@ -35,32 +35,35 @@ msgid ""
 ":class:`~optuna.study.Study`, and it is useful for deploying optimization"
 " results."
 msgstr ""
+"该对象有和 :class:`~optuna.trial.Trial` 一样的方法, 而且其提供预先定义好的参数值. 这些参数值可以"
+"在 :class:`~optuna.trial.FixedTrial` 的创建阶段指定. 和 :class:`~optuna.trial.Trial` 不同, "
+":class:`~optuna.trial.FixedTrial` 不依赖于 :class:`~optuna.study.Study`, 所以他对于部署优化结果是有用的."
```
Comment (belldandyxtq) [inline source/locale/zh_CN/LC_MESSAGES/reference/generated/optuna.trial.FixedTrial.po]: 所以能被用于部署优化结果.

File: source/locale/zh_CN/LC_MESSAGES/reference/generated/optuna.trial.FixedTrial.po [hunk 3]
```diff
@@ -35,32 +35,35 @@ msgid ""
...
 msgid "Evaluate an objective function with parameter values given by a user."
-msgstr ""
+msgstr "对用户提供的参数值进行目标函数求解."
 
 #: of optuna.trial.FixedTrial:27
 msgid ""
 "Please refer to :class:`~optuna.trial.Trial` for details of methods and "
 "properties."
-msgstr ""
+msgstr "细节和属性请参考 :class:`~optuna.trial.Trial`."
```
Comment (belldandyxtq) [inline source/locale/zh_CN/LC_MESSAGES/reference/generated/optuna.trial.FixedTrial.po]: 方法和属性的细节请参考

File: source/locale/zh_CN/LC_MESSAGES/reference/generated/optuna.trial.FrozenTrial.po [hunk 4]
```diff
@@ -33,88 +33,96 @@ msgid ""
 " depend on :class:`~optuna.study.Study`, and it is useful for deploying "
 "optimization results."
 msgstr ""
+"该对象有和 :class:`~optuna.trial.Trial` 一样的方法, 而且其提供预先定义好的参数值. 这些参数值可以"
+"在 :class:`~optuna.trial.FixedTrial` 的创建阶段指定. 和 :class:`~optuna.trial.Trial` 不同, "
+":class:`~optuna.trial.FrozenTrial` 不依赖于 :class:`~optuna.study.Study`, 所以他对于部署优化结果是有用的."
```
Comment (belldandyxtq) [inline source/locale/zh_CN/LC_MESSAGES/reference/generated/optuna.trial.FrozenTrial.po]: 所以能被用于部署优化结果.

File: source/locale/zh_CN/LC_MESSAGES/reference/generated/optuna.trial.FrozenTrial.po [hunk 5]
```diff
@@ -33,88 +33,96 @@ msgid ""
...
+msgstr "利用优化后的参数对一个目标函数重新求值."
 
 #: of optuna.trial._frozen.FrozenTrial:28
 msgid ""
 "Attributes are set in :func:`optuna.Study.optimize`, but several "
 "attributes can be updated after the optimization. That means such "
 "attributes are overwritten by the re-evaluation if your objective updates"
 " attributes of :class:`~optuna.trial.Trial`."
-msgstr ""
+msgstr "属性通过 :func:`optuna.Study.optimize` 来设定, 但是其中一些属性可以在优化后被更新. "
```
Comment (belldandyxtq) [inline source/locale/zh_CN/LC_MESSAGES/reference/generated/optuna.trial.FrozenTrial.po:54]: Please see https://github.com/optuna/optuna/pull/2833

File: source/locale/zh_CN/LC_MESSAGES/reference/generated/optuna.trial.FrozenTrial.po [hunk 6]
```diff
@@ -33,88 +33,96 @@ msgid ""
...
 msgid "Overwritten attributes."
-msgstr ""
+msgstr "覆盖属性."
 
 #: of optuna.trial._frozen.FrozenTrial:68
 msgid ""
 "Please refer to :class:`~optuna.trial.Trial` for details of methods and "
 "properties."
-msgstr ""
+msgstr "细节和属性请参考 :class:`~optuna.trial.Trial`."
```
Comment (belldandyxtq) [inline source/locale/zh_CN/LC_MESSAGES/reference/generated/optuna.trial.FrozenTrial.po]: 方法和属性的细节请参考

File: source/locale/zh_CN/LC_MESSAGES/reference/generated/optuna.trial.FrozenTrial.po [hunk 7]
```diff
@@ -33,88 +33,96 @@ msgid ""
...
 
 #: of optuna.trial._frozen.FrozenTrial:73
 msgid ""
 "Unique and consecutive number of :class:`~optuna.trial.Trial` for each "
 ":class:`~optuna.study.Study`. Note that this field uses zero-based "
 "numbering."
-msgstr ""
+msgstr "每一个 :class:`~optuna.study.Study` 所有的独特和连续的 :class:`~optuna.trial.Trial` 数字. "
```
Comment (belldandyxtq) [inline source/locale/zh_CN/LC_MESSAGES/reference/generated/optuna.trial.FrozenTrial.po]: 每一个 :class:`~optuna.study.Study` 所有的独特和连续的 :class:`~optuna.trial.Trial` 号.

File: source/locale/zh_CN/LC_MESSAGES/reference/generated/optuna.trial.FrozenTrial.po [hunk 8]
```diff
@@ -33,88 +33,96 @@ msgid ""
...
 msgid "Objective value of the :class:`~optuna.trial.Trial`."
-msgstr ""
+msgstr ":class:`~optuna.trial.Trial` 的目标函数值."
 
 #: of optuna.trial._frozen.FrozenTrial:86
 msgid ""
 "Sequence of objective values of the :class:`~optuna.trial.Trial`. The "
 "length is greater than 1 if the problem is multi-objective optimization."
-msgstr ""
+msgstr "Trial 目标函数值的序列. 如果该问题为多目标优化, 则其长度大于1. 如果 ``state`` 是 ``None`` 或者 "
```
Comment (belldandyxtq) [inline source/locale/zh_CN/LC_MESSAGES/reference/generated/optuna.trial.FrozenTrial.po]: I do not see the part after "如果" in the original text

File: source/locale/zh_CN/LC_MESSAGES/reference/generated/optuna.trial.FrozenTrial.po [hunk 9]
```diff
@@ -253,37 +261,39 @@ msgstr ""
...
+msgstr "从目标函数返回的值."
 
 #: of optuna.trial._frozen.FrozenTrial.report:10
 msgid ""
 "Step of the trial (e.g., Epoch of neural network training). Note that "
 "pruners assume that ``step`` starts at zero. For example, "
 ":class:`~optuna.pruners.MedianPruner` simply checks if ``step`` is less "
 "than ``n_warmup_steps`` as the warmup mechanism."
-msgstr ""
+msgstr "Trial 步骤（例如, 神经网络训练的 epoch. "
```
Comment (belldandyxtq) [inline source/locale/zh_CN/LC_MESSAGES/reference/generated/optuna.trial.FrozenTrial.po]: missing ) at the end

File: source/locale/zh_CN/LC_MESSAGES/reference/generated/optuna.trial.Trial.po [hunk 10]
```diff
@@ -39,6 +41,9 @@ msgid ""
 "behind the :func:`optuna.study.Study.optimize()` method; hence library "
 "users do not care about instantiation of this object."
 msgstr ""
+"请注意, 我们不建议直接使用此构造函数.  该对象被无缝实例化, 并传递给 "
+":func:`optuna.study.Study.optimize()` 方法后面的目标函数； "
+"因此, 库用户并不关心该对象的实例化. "
```
Comment (belldandyxtq) [inline source/locale/zh_CN/LC_MESSAGES/reference/generated/optuna.trial.Trial.po:46]: 因此, 库用户并不用关心该对象的实例化.

File: source/locale/zh_CN/LC_MESSAGES/reference/generated/optuna.trial.Trial.po [hunk 11]
```diff
@@ -298,7 +310,9 @@ msgid ""
 "pruners assume that ``step`` starts at zero. For example, "
 ":class:`~optuna.pruners.MedianPruner` simply checks if ``step`` is less "
 "than ``n_warmup_steps`` as the warmup mechanism."
-msgstr ""
+msgstr "Trial 步骤（例如, 神经网络训练的 epoch）. "
```
Comment (belldandyxtq) [inline source/locale/zh_CN/LC_MESSAGES/reference/generated/optuna.trial.Trial.po]: I don't think the first part matches the English

`目标函数的返回值`

File: source/locale/zh_CN/LC_MESSAGES/reference/generated/optuna.trial.Trial.po [hunk 12]
```diff
@@ -329,51 +343,58 @@ msgid ""
 "Note that Optuna internally uses this method to save system messages such"
 " as failure reason of trials. Please use "
 ":func:`~optuna.trial.Trial.set_user_attr` to set users' attributes."
-msgstr ""
+msgstr "注意, optuna 在内部使用该方法来保存如 trial 失败之类的系统信息. 请使用"
```
Comment (belldandyxtq) [inline source/locale/zh_CN/LC_MESSAGES/reference/generated/optuna.trial.Trial.po]: ... 保存如 trial 失败原因之类的系统信息.

File: source/locale/zh_CN/LC_MESSAGES/reference/generated/optuna.trial.Trial.po [hunk 13]
```diff
@@ -329,51 +343,58 @@ msgid ""
...
 msgid "A value of the attribute. The value should be JSON serializable."
-msgstr ""
+msgstr "属性的值.  该值应为JSON可序列化的. "
 
 #: of optuna.trial._trial.Trial.set_user_attr:3
 msgid ""
 "The user attributes in the trial can be access via "
 ":func:`optuna.trial.Trial.user_attrs`."
-msgstr ""
+msgstr "可以通过 :func:`optuna.trial.Trial.user_attrs` 访问试 trial 中的用户属性. "
```
Comment (belldandyxtq) [inline source/locale/zh_CN/LC_MESSAGES/reference/generated/optuna.trial.Trial.po]: 访问~试~

File: source/locale/zh_CN/LC_MESSAGES/reference/generated/optuna.trial.Trial.po [hunk 14]
```diff
@@ -384,6 +405,9 @@ msgid ""
 "Suggest a kernel function of `SVC <https://scikit-"
 "learn.org/stable/modules/generated/ sklearn.svm.SVC.html>`_."
 msgstr ""
+"为 `SVC <https://scikit-"
```
Comment (belldandyxtq) [inline source/locale/zh_CN/LC_MESSAGES/reference/generated/optuna.trial.Trial.po:408]: missing translation before this
`该值从choices中取样`

File: source/locale/zh_CN/LC_MESSAGES/reference/generated/optuna.trial.Trial.po [hunk 15]
```diff
@@ -392,19 +416,19 @@ msgstr ""
...
+msgstr "参数值候选."
 
 #: of optuna.trial._trial.Trial.suggest_categorical:37
 msgid ":class:`~optuna.distributions.CategoricalDistribution`."
 msgstr ""
 
 #: of optuna.trial._trial.Trial.suggest_categorical:39
 msgid "A suggested value."
-msgstr ""
+msgstr "suggesttion 值."
```
Comment (belldandyxtq) [inline source/locale/zh_CN/LC_MESSAGES/reference/generated/optuna.trial.Trial.po]: suggestion

typo

File: source/locale/zh_CN/LC_MESSAGES/reference/generated/optuna.trial.Trial.po [hunk 16]
```diff
@@ -470,12 +501,15 @@ msgid ""
 ":func:`~optuna.trial.Trial.suggest_loguniform` and "
 ":func:`~optuna.trial.Trial.suggest_discrete_uniform`."
 msgstr ""
+"参见 :func:`~optuna.trial.Trial.suggest_uniform`, "
+":func:`~optuna.trial.Trial.suggest_loguniform` and "
```
Comment (belldandyxtq) [inline source/locale/zh_CN/LC_MESSAGES/reference/generated/optuna.trial.Trial.po]: and -> 和

File: source/locale/zh_CN/LC_MESSAGES/reference/generated/optuna.trial.Trial.po [hunk 17]
```diff
@@ -532,30 +569,42 @@ msgid ""
...
 "``None``."
 msgstr ""
+"``step`` 和 ``log`` 参数不能同时使用. 要把 ``step`` "
+"设置成浮点数的话, 请先把 ``log`` 参数设置成 ``False``."
+
 
 #: of optuna.trial._trial.Trial.suggest_float:76
 msgid "If ``step is not None`` and ``log = True`` are specified."
-msgstr ""
+msgstr "如果 ``step is not None`` 并且制定了 ``log = True``."
```
Comment (belldandyxtq) [inline source/locale/zh_CN/LC_MESSAGES/reference/generated/optuna.trial.Trial.po]: 如果 ``step is not None`` 并且指定了 ``log = True``

File: source/locale/zh_CN/LC_MESSAGES/reference/generated/optuna.trial.Trial.po [hunk 18]
```diff
@@ -576,15 +625,20 @@ msgid ""
 "Note that :math:`\\mathsf{high}` is modified if the range is not "
 "divisible by :math:`\\mathsf{step}`. Please check the warning messages to"
 " find the changed values."
-msgstr ""
+msgstr "注意,  "
```
Comment (belldandyxtq) [inline source/locale/zh_CN/LC_MESSAGES/reference/generated/optuna.trial.Trial.po:631]: Missing translation at line 611?

File: source/locale/zh_CN/LC_MESSAGES/reference/generated/optuna.trial.Trial.po [hunk 19]
```diff
@@ -623,47 +677,61 @@ msgid ""
...
+" `low = 2` 且 `high = 8` "
+"的话,  suggestion 值的范围就是 `[2, 3, 4, "
+"5, 6, 7, 8]` 而且较小的值有更大的概率被采样到. "
 
 #: of optuna.trial._trial.Trial.suggest_int:66
 msgid ""
 "The ``step != 1`` and ``log`` arguments cannot be used at the same time. "
 "To set the ``log`` argument to ``True``, set the ``step`` argument to 1."
 msgstr ""
+"``step != 1`` 和 ``log`` 参数不能同时使用要把 "
```
Comment (belldandyxtq) [inline source/locale/zh_CN/LC_MESSAGES/reference/generated/optuna.trial.Trial.po]: 参数不能同时使用。要把

**Review by belldandyxtq — CHANGES_REQUESTED** [review]

File: source/locale/zh_CN/LC_MESSAGES/reference/generated/optuna.trial.FixedTrial.po (hunk 1 above)
Comment (chenghuzi) [reply source/locale/zh_CN/LC_MESSAGES/reference/generated/optuna.trial.FixedTrial.po]: Could it be "为每个参数都提供固定值的 trial 类."?

File: source/locale/zh_CN/LC_MESSAGES/reference/generated/optuna.trial.Trial.po (hunk 11 above)
Comment (chenghuzi) [reply source/locale/zh_CN/LC_MESSAGES/reference/generated/optuna.trial.Trial.po]: I don't understand this part, why "目标函数的返回值" is here?

File: source/locale/zh_CN/LC_MESSAGES/reference/generated/optuna.trial.Trial.po (hunk 18 above)
Comment (chenghuzi) [reply source/locale/zh_CN/LC_MESSAGES/reference/generated/optuna.trial.Trial.po:631]: No this is reflected in line 629. Any translation here leads to crashed rendered HTML output.

File: source/locale/zh_CN/LC_MESSAGES/reference/generated/optuna.trial.FrozenTrial.po (hunk 5 above)
Comment (chenghuzi) [reply source/locale/zh_CN/LC_MESSAGES/reference/generated/optuna.trial.FrozenTrial.po:54]: Will finish this part once we update the original doc to the latest version.

File: source/locale/zh_CN/LC_MESSAGES/reference/generated/optuna.trial.Trial.po (hunk 11 above)
Comment (belldandyxtq) [reply source/locale/zh_CN/LC_MESSAGES/reference/generated/optuna.trial.Trial.po]: I think the text has shifted somehow, the `目标函数的返回值` is the translation above at line 305

**Review by belldandyxtq — APPROVED** [review]

**Pull request merged**